### PR TITLE
jigasi,web: add transcriber options

### PR DESCRIPTION
**Pull request by netaskd**
The PR adds opportunity to enable transcribing in a conference.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +121,17 @@ Variable | Description | Example
 `JIGASI_SIP_PORT` | SIP server port | 5060
 `JIGASI_SIP_TRANSPORT` | SIP transport | UDP
 
+### Transcription configuration
```
Comment (saghul) [inline README.md]: Please move this after the authentication section.

File: README.md [hunk 2]
```diff
@@ -121,6 +121,17 @@ Variable | Description | Example
...
+
+If you want to enable the Transcribing function, these options are required:
+
+Variable | Description | Example
+--- | --- | ---
+`ENABLE_JIGASI_TRANSCRIBER` | Enable Jigasi transcription in a conference | 1
+`GOOGLE_APPLICATION_CREDENTIALS` | Credentials for connect to Cloud Google API from Jigasi. Path located inside the container | /config/key.json
+
+For set `GOOGLE_APPLICATION_CREDENTIALS` please read https://cloud.google.com/text-to-speech/docs/quickstart-protocol section "Before you begin" from 1 to 5 paragraph
```
Comment (saghul) [inline README.md]: "For setting", also end the sentence in a period.

File: README.md [hunk 3]
```diff
@@ -121,6 +121,17 @@ Variable | Description | Example
 `JIGASI_SIP_PORT` | SIP server port | 5060
 `JIGASI_SIP_TRANSPORT` | SIP transport | UDP
 
+### Transcription configuration
+
+If you want to enable the Transcribing function, these options are required:
+
+Variable | Description | Example
+--- | --- | ---
+`ENABLE_JIGASI_TRANSCRIBER` | Enable Jigasi transcription in a conference | 1
```
Comment (saghul) [inline README.md]: Let's call this `ENABLE_TRANSCRIPTIONS` please.

File: jigasi/rootfs/defaults/sip-communicator.properties [hunk 4]
```diff
@@ -103,3 +103,19 @@ org.jitsi.jigasi.xmpp.acc.ALLOW_NON_SECURE=true
 # type of non-trusted certicates. In this mode your service trust in the
 # remote certificates always.
 net.java.sip.communicator.service.gui.ALWAYS_TRUST_MODE_ENABLED=true
+
+{{ if .Env.ENABLE_JIGASI_TRANSCRIBER | default "0" | toBool }}
+# Transcription config
+org.jitsi.jigasi.ENABLE_TRANSCRIPTION=true
+org.jitsi.jigasi.transcription.ENABLE_TRANSLATION=true
+org.jitsi.jigasi.transcription.DIRECTORY=/tmp/transcripts
+org.jitsi.jigasi.transcription.BASE_URL=https://{{ .Env.LETSENCRYPT_DOMAIN }}/transcripts
```
Comment (saghul) [inline jigasi/rootfs/defaults/sip-communicator.properties]: this won't work when the setup is behind a proxy and not using the builtin let's encrypt facility. I think we may need some PUBLIC_URL setting.

File: jigasi/rootfs/etc/cont-init.d/10-config [hunk 5]
```diff
@@ -7,3 +7,6 @@ fi
 if [[ ! -f /config/logging.properties ]]; then
     cp /defaults/logging.properties /config
 fi
+
+mkdir -pm777 /tmp/transcripts
+chown jigasi.jitsi /tmp/transcripts
```
Comment (saghul) [inline jigasi/rootfs/etc/cont-init.d/10-config]: IIRC it was recommended to use the `:` notation instead.

**Comment by netaskd** [comment]
Hi @saghul,
Sorry for delay. All conversations are resolved.

File: jigasi.yml [hunk 6]
```diff
@@ -24,6 +26,12 @@ services:
             - JIGASI_BREWERY_MUC
             - JIGASI_PORT_MIN
             - JIGASI_PORT_MAX
+            - ENABLE_TRANSCRIPTIONS
+            - JIGASI_TRANSCRIBER_ADVERTISE_URL
+            - JIGASI_TRANSCRIBER_RECORD_AUDIO
+            - JIGASI_TRANSCRIBER_SEND_TXT
+            - GOOGLE_APPLICATION_CREDENTIALS
+            - LETSENCRYPT_DOMAIN
```
Comment (saghul) [inline jigasi.yml]: is this one still needed then?

Comment (netaskd) [reply jigasi.yml]: oh, missed it. I removed line `- LETSENCRYPT_DOMAIN`

File: env.example [hunk 7]
```diff
@@ -14,6 +14,9 @@ HTTPS_PORT=8443
 # System time zone.
 TZ=Europe/Amsterdam
 
+# Public domain name of the web service.
+PUBLIC_DOMAIN="meet.example.com"
```
Comment (saghul) [inline env.example]: please leave it commented out and move it to the advanced section, also I think we can remove the quotes

File: README.md [hunk 8]
```diff
@@ -91,6 +91,7 @@ Variable | Description | Example
 `HTTP_PORT` | Exposed port for HTTP traffic | 8000
 `HTTPS_PORT` | Exposed port for HTTPS traffic | 8443
 `DOCKER_HOST_ADDRESS` | IP address of the Docker host, needed for LAN environments | 192.168.1.1
+`PUBLIC_DOMAIN` | Public domain name of the web service | meet.example.com
```
Comment (saghul) [inline README.md]: let's call this PUBLIC_URL, and take a URL instead of a domain.

File: jigasi/rootfs/defaults/sip-communicator.properties [hunk 9]
```diff
@@ -103,3 +103,19 @@ org.jitsi.jigasi.xmpp.acc.ALLOW_NON_SECURE=true
 # type of non-trusted certicates. In this mode your service trust in the
 # remote certificates always.
 net.java.sip.communicator.service.gui.ALWAYS_TRUST_MODE_ENABLED=true
+
+{{ if .Env.ENABLE_TRANSCRIPTIONS | default "0" | toBool }}
+# Transcription config
+org.jitsi.jigasi.ENABLE_TRANSCRIPTION=true
+org.jitsi.jigasi.transcription.ENABLE_TRANSLATION=true
+org.jitsi.jigasi.transcription.DIRECTORY=/tmp/transcripts
+org.jitsi.jigasi.transcription.BASE_URL=https://{{ .Env.PUBLIC_DOMAIN }}/transcripts
```
Comment (saghul) [inline jigasi/rootfs/defaults/sip-communicator.properties]: this needs adjusting when changing to PUBLIC_URL

**Comment by saghul** [comment]
Sorry for the delay, I've been quite swamped lately. Left some last comments.

File: env.example [hunk 10]
```diff
@@ -14,6 +14,9 @@ HTTPS_PORT=8443
 # System time zone.
 TZ=Europe/Amsterdam
 
+# Public URL for the web service.
+PUBLIC_URL="https://meet.example.com"
```
Comment (saghul) [inline env.example]: please leave this one commented out, as the user needs to edit it

Comment (netaskd) [reply env.example]: Ok

**Comment by saghul** [comment]
Rebased, squashed and landed in ✨  e48bf46 ✨ , thank you!

**Pull request closed without merging**